src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -320,6 +326,145 @@ def cli_cosmosdb_sql_container_update(client,
                                               sql_container_create_update_resource)
 
 
+def cli_cosmosdb_sql_stored_procedure_create_update(client,
+                                                    resource_group_name,
+                                                    account_name,
+                                                    database_name,
+                                                    container_name,
+                                                    stored_procedure_name,
+                                                    stored_procedure_body):
+
+    """Creates or Updates an Azure Cosmos DB SQL stored procedure """
+    sql_stored_procedure_resource = SqlStoredProcedureResource(id=stored_procedure_name)
+    sql_stored_procedure_resource.body = stored_procedure_body
+
+    sql_stored_procedure_create_update_resource = SqlStoredProcedureCreateUpdateParameters(
+        resource=sql_stored_procedure_resource,
+        options={})
+
+    return client.create_update_sql_stored_procedure(resource_group_name,
```

Comment:
It looks this operation is a long running operation. We will recommend to support no wait for such operation and add an additional wait command to query operation status. Please refer to https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/network/commands.py#L351